test/UnitTests/MSTest.Analyzers.UnitTests/NonNullableReferenceNotInitializedSuppressorTests.cs

```diff
@@ -54,56 +42,55 @@ public async Task TestContextPropertyOnNonTestClass_DiagnosticIsNotSuppressed()
-    [SuppressMessage("MicrosoftCodeAnalysisCorrectness", "RS1036:Specify analyzer banned API enforcement setting", Justification = "For suppression test only.")]
-    [SuppressMessage("MicrosoftCodeAnalysisCorrectness", "RS1041:Compiler extensions should be implemented in assemblies targeting netstandard2.0", Justification = "For suppression test only.")]
-    public class DoNothingAnalyzer : DiagnosticAnalyzer
+    [TestMethod]
+    public async Task TestContextPropertyOnTestClassConstructor_DiagnosticIsSuppressed()
     {
-        [SuppressMessage("MicrosoftCodeAnalysisDesign", "RS1017:DiagnosticId for analyzers must be a non-null constant.", Justification = "For suppression test only.")]
-        public static readonly DiagnosticDescriptor Rule = new(NonNullableReferenceNotInitializedSuppressor.Rule.SuppressedDiagnosticId, "Title", "Message", "Category", DiagnosticSeverity.Warning, isEnabledByDefault: true);
+        string code = @"
+#nullable enable
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => [Rule];
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
 
-        public override void Initialize(AnalysisContext context)
-        {
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
-            context.EnableConcurrentExecution();
+[TestClass]
+public class SomeClass
+{
+    public {|#0:SomeClass|}()
+    {
+    }
 
-            context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Property);
-        }
+    public TestContext TestContext { get; set; }
+}
+";
 
-        private void AnalyzeSymbol(SymbolAnalysisContext context)
-        {
-        }
+        await VerifySingleSuppressionAsync(code, true);
     }
 
-    internal sealed class TestWithSuppressor : VerifyCS.Test
+    private Task VerifySingleSuppressionAsync(string source, bool isSuppressed)
+        => VerifyDiagnosticsAsync(source, [(0, isSuppressed)]);
+
+    private async Task VerifyDiagnosticsAsync(string source, List<(int Location, bool IsSuppressed)> diagnostics)
     {
-        protected override IEnumerable<DiagnosticAnalyzer> GetDiagnosticAnalyzers()
+        var test = new VerifyCS.Test
         {
-            foreach (DiagnosticAnalyzer analyzer in base.GetDiagnosticAnalyzers())
-            {
-                yield return analyzer;
-            }
+            TestCode = source,
+        };
 
-            yield return new NonNullableReferenceNotInitializedSuppressor();
+        foreach ((int location, bool isSuppressed) in diagnostics)
+        {
+            test.ExpectedDiagnostics.Add(DiagnosticResult.CompilerError("CS8618")
+                .WithLocation(location)
+                .WithOptions(DiagnosticOptions.IgnoreAdditionalLocations)
+                .WithArguments("property", "TestContext")
+                .WithIsSuppressed(isSuppressed));
```

Comment:
We need to update all other suppressor tests similar to here. I'll track that separately.

On the old form, the tests are not asserting anything, and commented out the `context.ReportSuppression` call will still pass all tests.